src/lib/lwan.c

```diff
@@ -132,6 +132,42 @@ static struct lwan_url_map *add_url_map(struct lwan_trie *t, const char *prefix,
     return copy;
 }
 
+static void parse_global_headers(struct config *c,
+                                 const struct config_line *l,
+                                 struct lwan *lwan,
+                                 struct lwan_key_value_array * hdrs)
+{
+    assert(hdrs);
+
+    struct lwan_key_value * kv;
+    lwan_key_value_array_init(hdrs);
+
+    while ((l = config_read_line(c))) {
+        switch (l->type) {
+        case CONFIG_LINE_TYPE_SECTION:
+            config_error(c, "No sections are supported under the headers section");
+            goto error;
+
+        case CONFIG_LINE_TYPE_LINE:
+            kv = lwan_key_value_array_append(hdrs);
```

Comment:
`lwan_key_value_array_append()` can return NULL if allocation fails.

```suggestion
            kv = lwan_key_value_array_append(hdrs);
            if (!kv)
            	lwan_status_critical_perror("Could not allocate memory for custom response header");
```